### ISSUE TITLE
Null Exception When Archetype Property is Missing

### ISSUE THREAD

**Issue by Nicholas-Westby** [micklaw/Ditto.Resolvers#12]
When I map a model using Ditto and Ditto Resolvers, I get a null exception with this stack trace:

```
   at Our.Umbraco.Ditto.Resolvers.Archetype.Resolvers.ArchetypeValueResolver.ResolveValue()
   at Our.Umbraco.Ditto.DittoValueResolver`2.ResolveValue(DittoValueResolverContext context, DittoValueResolverAttribute attribute, CultureInfo culture)
   at Our.Umbraco.Ditto.PublishedContentExtensions.GetResolvedValue(IPublishedContent content, CultureInfo culture, PropertyInfo propertyInfo, Object instance, IEnumerable`1 valueResolverContexts)
   at Our.Umbraco.Ditto.PublishedContentExtensions.ConvertContent(IPublishedContent content, Type type, CultureInfo culture, Object instance, IEnumerable`1 valueResolverContexts, Action`1 onConverting, Action`1 onConverted)
   at Our.Umbraco.Ditto.PublishedContentExtensions.As(IPublishedContent content, Type type, CultureInfo culture, Object instance, IEnumerable`1 valueResolverContexts, Action`1 onConverting, Action`1 onConverted)
   at Our.Umbraco.Ditto.PublishedContentExtensions.As[T](IPublishedContent content, CultureInfo culture, T instance, IEnumerable`1 valueResolverContexts, Action`1 onConverting, Action`1 onConverted)
   at MySite.App.Converters.QuadActionsConverter.AssureQuadActionItems(IPublishedContent page, List`1 actions) in C:\r\MySite\src\MySite.App\Converters\QuadActionsConverter.cs:line 68
   at MySite.App.Converters.QuadActionsConverter.ConvertFrom(ITypeDescriptorContext context, CultureInfo culture, Object value) in C:\r\MySite\src\MySite.App\Converters\QuadActionsConverter.cs:line 47
```

Here is the line throwing the error:

```
var quadContainer = page.As<QuadActionContainer>();
```

Here's what that model class looks like:

```
public class QuadActionContainer
{
    [ArchetypeValueResolver]
    public List<IQuadActionItem> QuadActions { get; set; }
}
```

The problem seems to be that the content node I'm mapping doesn't happen to have a property with an alias of `QuadActions` on it, which is what I happen to expect of this page.

What I'd like is for the `QuadActions` property to be set to null rather than throw an exception. It would be even better if it were set to an empty list, but beggars can't be choosers :-)

**Comment by micklaw** [comment]
@Nicholas-Westby, try that 1.0.5 release. It may be fixed in that. In my sample project I have added a random property to both POCOS and both worked ok (nullable)

I just added a some new functionality to the dev branch where these are created as new lists. I'll upload the dlls when the build completes

**Comment by micklaw** [comment]
[Ditto.Resolvers.Archetype.1.0.6-alpha-000139.zip](https://github.com/micklaw/Ditto.Resolvers/files/122339/Ditto.Resolvers.Archetype.1.0.6-alpha-000139.zip)

**Comment by Nicholas-Westby** [comment]
Yep, 1.0.5 is now returning null as I'd expect! I'll wait for whenever your next release comes out for the empty list change :-)

**Comment by micklaw** [comment]
Awesome, il package it up and release it

**Comment by micklaw** [comment]
I release this fix as 1.0.6 dude
